### PR TITLE
[EDU-1526] Ensure full code coverage for go

### PR DESCRIPTION
**Pull request by GregHolmes**
## Description

Ensuring documentation has full code coverage for Go.

* [Jira epic ticket](https://ably.atlassian.net/browse/EDU-1489)
* [Jira ticket](https://ably.atlassian.net/browse/EDU-1526).

### CODE REVIEW

File: content/auth/capabilities.textile [hunk 1]
```diff
@@ -137,6 +164,35 @@ token = await ably.auth.create_token_request(
...
+//   "status": ["subscribe"]
+// }
+
+rest, _ := ably.NewREST(
+  ably.WithKey("{{API_KEY}}"))
+
+// Define the token parameters
+tokenParams := &ably.TokenParams{
+  ClientID: "client@example.com",
+  TTL:      3600 * 1000, // ms
```
Comment (sacOO7) [inline content/auth/capabilities.textile:180]: You can add 1 hr as a comment

**Review by sacOO7 — CHANGES_REQUESTED** [review]
Overall looks good to me. 
I assume every piece of code is tested independently in a separate repo

File: content/storage-history/history.textile [hunk 2]
```diff
@@ -416,6 +456,35 @@ channel.on(.attached) { error in
...
+err = channel.Publish(context.Background(), "example", "message data")
+if err != nil {
+  log.Fatalf("Failed to publish message: %v", err)
+}
+
+// Attach the channel
+_ = channel.Attach(context.Background())
+
+// Get channel history until attach
+history, _ := channel.HistoryUntilAttach()
```
Comment (sacOO7) [inline content/storage-history/history.textile:479]: Have we tested this explicitly in a separate example repo?
We don't have tests available for the same and only method was added as a part of protocol 2 work.
You can check method doc to test it explicitly.

**Review by m-hulbert — COMMENTED** [review]
Missing Go (and Python) as defined languages on capabilities.textile 🙂

File: content/storage-history/history.textile (hunk 2 above)
Comment (GregHolmes) [reply content/storage-history/history.textile:479]: I have tested every code snippet that I've added. But I haven't got a separate example repo for testing. I did this with PHP, but it was too time consuming to do for every language.

File: content/auth/token.textile [hunk 3]
```diff
@@ -569,6 +689,63 @@ $base64Signature = base64_encode($signature);
 $jwt = $base64Header . '.' . $base64Claims . '.' . $base64Signature;
 ```
 
+```[go]
```
Comment (sacOO7) [inline content/auth/token.textile:692]: This is a code representing server side construction of `jwt` token. 
I would also recommend adding client side code that uses `jwt` token using `authCallback`.
You can refer to https://github.com/ably/ably-dotnet?tab=readme-ov-file#using-the-token-auth-at-client-side

File: content/auth/token.textile [hunk 4]
```diff
@@ -480,6 +557,49 @@ $signature = hash_hmac(
 $jwt = $base64Header . '.' . $base64Claims . '.' . $signature;
 ```
 
+```[go]
```
Comment (sacOO7) [inline content/auth/token.textile:560]: This is a code representing server side construction of `jwt` token. 
I would also recommend adding client side code that uses `jwt` token using `authCallback`.
You can refer to https://github.com/ably/ably-dotnet?tab=readme-ov-file#using-the-token-auth-at-client-side

Comment (sacOO7) [reply content/auth/token.textile:560]: Using `jwt` at client side is tricky and depends on given SDK. 
That's why I strongly feel we include client code for each SDK that uses JWT string token using `authCallback`

**Review by sacOO7 — CHANGES_REQUESTED** [review]

File: content/storage-history/history.textile (hunk 2 above)
Comment (sacOO7) [reply content/storage-history/history.textile:479]: No issues, I assume you have tested `HistoryUntilAttach` that returns message history before attach only

File: content/auth/token.textile (hunk 4 above)
Comment (GregHolmes) [reply content/auth/token.textile:560]: I don't think that is an update for this pull request, but rather similar to the PHP PR, an addition that can improve the authentication section. Something we have in the pipeline. I've added this to the Authentication documentation update ticket: https://ably.atlassian.net/browse/EDU-1555

Comment (sacOO7) [reply content/auth/token.textile:560]: Okay, I can see it as a comment . 
Please do make sure this is addressed in the future for all SDKs 👍

File: content/auth/token.textile (hunk 3 above)
Comment (sacOO7) [reply content/auth/token.textile:692]: closing in favor of https://github.com/ably/docs/pull/2201#discussion_r1662291666

**Review by sacOO7 — APPROVED** [review]
LGTM 👍

**Review by m-hulbert — APPROVED** [review]
🚀

**Pull request merged**